internal/gengapic/gengapic.go

```diff
@@ -108,6 +111,10 @@ func Gen(genReq *plugin.CodeGeneratorRequest) (*plugin.CodeGeneratorResponse, er
 		return &g.resp, errors.E(nil, paramError)
 	}
 
+	if g.modulePrefix != "" && strings.HasPrefix(outDir, g.modulePrefix) {
```

Comment:
What about to handle wrong prefix like here: https://go-review.googlesource.com/c/protobuf/+/219298/4/compiler/protogen/protogen.go#417

Reply:
Done.